src/model/processor.py

```diff
@@ -0,0 +1,17 @@
+"""
+Abstract base class to augment data
+"""
+from abc import ABC
+
+
+class Processor(ABC):
+    """
+    Interface for a general approach to augment different artifacts
+    """
+    pass
+
+    def process(self, input_file, output_file):
```

Comment:
```suggestion
    def process(self, input_file_path, output_file_path):
```

Reply:
I agree to the renaming. However this should not necessarily be a string. You can also pass ready Path objects in Python which are e.g. operating system agnostic

Reply:
yea but the input from the user is a string, so a string would be the most fitting.

Reply:
But this is something the Command Line Interface should handle imho: It converts the string to a path and if e.g. the file does not exist it already tells the user, without even loading the Processor Object. If we just pass on the string (e.g. for the output_file_path) and in the end, when we want to write the output, we recognize we do not have permission or similar problems.

Reply:
i understand and agree. i looked for a "path" type, but cant find one without importing some additional library. if we use some additional library to do so, lets declare it now (example: from os import path or from pathlib import Path)

Reply:
> i looked for a "path" type, but cant find one without importing some additional library.

Well yeah you need to import one, but it's not really a additional library that one would need to install, as it is always provided with the standard python installation.

> lets declare it now (example: "from os import path" or "from pathlib import Path")

I disagree. Why would we need to artificially limit it now? This depends on what is supported by the CLI-framework and the library we use in the processors. So we can still limit it later if necessary.